contracts/multisig-prover/src/encoding/abi.rs

```diff
@@ -1,189 +1,138 @@
 use std::str::FromStr;
 
-use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{HexBinary, Uint256};
-use ethabi::{ethereum_types, short_signature, ParamType, Token};
+use ethabi::{short_signature, ParamType, Token};
 use k256::{elliptic_curve::sec1::ToEncodedPoint, PublicKey};
-use sha3::{Digest, Keccak256};
-
-use connection_router::msg::Message;
 use multisig::{key::Signature, msg::Signer};
+use sha3::{Digest, Keccak256};
 
 use crate::{
     error::ContractError,
     state::WorkerSet,
-    types::{BatchID, Command, CommandBatch, CommandType, Operator},
+    types::{CommandBatch, Operator},
 };
 
-const GATEWAY_EXECUTE_FUNCTION_NAME: &str = "execute";
-
-impl TryFrom<Message> for Command {
-    type Error = ContractError;
-
-    fn try_from(msg: Message) -> Result<Self, Self::Error> {
-        Ok(Command {
-            ty: CommandType::ApproveContractCall, // TODO: this would change when other command types are supported
-            params: command_params(
-                msg.source_chain,
-                msg.source_address,
-                ethereum_types::Address::from_str(&msg.destination_address).map_err(|e| {
-                    ContractError::InvalidMessage {
-                        reason: format!("destination_address is not a valid EVM address: {}", e),
-                    }
-                })?,
-                msg.payload_hash.as_slice().try_into().map_err(|e| {
-                    ContractError::InvalidMessage {
-                        reason: format!(
-                            "payload_hash length is not a valid keccak256 hash length: {}",
-                            e
-                        ),
-                    }
-                })?,
-            ),
-            id: command_id(msg.id),
-        })
-    }
-}
+use super::Data;
+use ethabi::ethereum_types;
 
-fn make_transfer_operatorship(worker_set: WorkerSet) -> Result<Command, ContractError> {
-    let params = transfer_operatorship_params(&worker_set)?;
-    Ok(Command {
-        ty: CommandType::TransferOperatorship,
-        params,
-        id: worker_set.hash(),
-    })
-}
+pub const GATEWAY_EXECUTE_FUNCTION_NAME: &str = "execute";
 
-impl TryFrom<Signer> for Operator {
-    type Error = ContractError;
+pub(super) fn encode(data: &Data) -> HexBinary {
```

Comment:
What's the thought here for making this function visible in the parent mod?

Reply:
the parent mod has `pub fn encode(&self, encoding: EncodingLanguage)` function that calls this function. Other mods should just call the parent mod function.

Reply:
But do you need to make this `super` here to do that? The parent only always import the child, no?

Reply:
if he wouldn't the parent couldn't call `ecode`. Children can call private functions of their parents, not the other way around

Reply:
@cgorenflo What I was trying to say is that the child doesn't seem to have to be exported as a pub mod in the parent, and then these functions here can just be `pub` instead of `pub(super)`. It looks good now though.